build/lib/pgBackRestBuild/Config/Data.pm

```diff
@@ -16,9 +16,15 @@
 #         (B), else it will still error on the option (A).
 #
 # CFGDEF_COMMAND_ROLE:
+#     1) Define the command roles for a command. CFGCMD_ROLE_DEFAULT is valid for all commands and is therefore added
+#        programmatically.
 #
-# Define the command roles that a command is valid for. CFGCMD_ROLE_DEFAULT is valid for all commands and is therefore added
-# programmatically.
+#     2) Define the command roles for an option. CFGCMD_ROLE_DEFAULT must be defined explictly in this case. If command roles are
+#        defined for the option they override the default of all roles in the command.
```

Comment:
So, let's take EXPIRE command for an example. It does not have a role defined so it programmatically gets the CFGCMD_ROLE_DEFAULT. Then the option DRYRUN has only the EXPIRE command listed so it's the only command it is valid for. So when specifying the dryrun option then the expire command is only allowed to be in the default mode?  Maybe backup is a better example. BACKUP has local and remote (and default) command roles but ONLINE option has only the default command role so when the backup command is run with the online option then the default role overrides the local and remote roles of the backup command which means, what, backup can't run in local or remote roles? Maybe we can talk this and definition 3) out; with a bit better picture, I might be able to clarify.

Reply:
Fixed a1dacac05867593bba449e89731efcadb90d7405 (per discussion with David)